trend-analysis/TrendAnalysisApp.UnitTests/TrendDataProcessorTests.cs

```diff
@@ -45,14 +81,258 @@ public void TestUpdateIndicatorSearchDataWritesCorrectDataToFile() {
+        {
+            IndicatorKey = gpRegIndicatorKey,
+            IndicatorId = Indicator.GpRegisteredPopulationId,
+        });
+        var residentPopulationIndicator = _dbContext.IndicatorDimension.Add(new IndicatorDimensionModel
+        {
+            IndicatorKey = residentPopulationIndicatorKey,
+            IndicatorId = Indicator.ResidentPopulationId,
+        });
+        _dbContext.SaveChanges();
+
+        PopulateMockHealthMeasureData(gpRegIndicator.Entity.IndicatorKey);
+        PopulateMockHealthMeasureData(residentPopulationIndicator.Entity.IndicatorKey, DefaultAreaEntityKey, residentPopulationIndicatorKey);
+        
+        // act
+        await _trendDataProcessor.Process(_serviceProvider);
+
+        // extract
+        var actualGpReg = _dbContext.HealthMeasure.Find((int)gpRegIndicatorKey);
+        var actualResPop = _dbContext.HealthMeasure.Find((int)residentPopulationIndicatorKey);
+
+        // assert
+        actualGpReg.TrendDimension.Name.ShouldBe(Constants.Trend.NotYetCalculated);
+        actualResPop.TrendDimension.Name.ShouldBe(Constants.Trend.NotYetCalculated);
+        _mockFileHelper.Received().Read(ExpectedFilePath);
+        _mockFileHelper.Received().Write(
+            ExpectedFilePath,
+            Arg.Is<JArray>(ja => JToken.DeepEquals(ja, mockFileContents))
+        );
+
+    }
+
+    [Fact]
+    public async Task TestTrendDataProcessor_CalculatesTrendsForRelevantGroupedData()
+    {
+        // arrange
+        const short mockIndicatorKey = 1;
+
+        _mockFileHelper.Read(ExpectedFilePath).Returns(
+            new JArray { { new JObject { ["indicatorID"] = mockIndicatorKey } } }
+        );
+        JArray expectedUpdatedFileContents = new()
+        {
+            {
+                new JObject
+                {
+                    ["indicatorID"] = mockIndicatorKey, ["trendsByArea"] = new JArray
+                    {
+                        new JObject
+                        {
+                            ["areaCode"] = "AC1",
+                            ["trend"] = Constants.Trend.NoSignificantChange
+                        },
+                        new JObject
+                        {
+                            ["areaCode"] = "AC2",
+                            ["trend"] = Constants.Trend.CannotBeCalculated
+                        }
+                    }
+                }
+            }
+        };
+
+        var mockIndicator = _dbContext.IndicatorDimension.Add(new IndicatorDimensionModel
+        {
+            IndicatorKey = mockIndicatorKey,
+            IndicatorId = 1,
+            Polarity = Polarity.NoJudgement,
+            ValueType = "Directly standardised rate"
+        });
+        _dbContext.SaveChanges();
+
+        PopulateMockHealthMeasureData(mockIndicator.Entity.IndicatorKey);
+        PopulateMockHealthMeasureData(mockIndicator.Entity.IndicatorKey, DefaultAreaEntityKey+1);
```

Comment:
```suggestion
        PopulateMockHealthMeasureData(mockIndicator.Entity.IndicatorKey, DefaultAreaEntityKey + 1);
```

As an aside, it would probably be nice to add a new const for the second area key and maybe adjust the default area key variable name to be something like areaOneKey. Something like that anyway.